compiler/parser/chapel.lex

```diff
@@ -468,6 +487,63 @@ static const char* eatStringLiteral(yyscan_t scanner, const char* startChar) {
   return astr(stringBuffer);
 }
 
+static const char* eatMultilineStringLiteral(yyscan_t scanner,
+                                             const char* startChar) {
+  YYLTYPE*   yyLloc  = yyget_lloc(scanner);
+  const char startCh = *startChar;
+  int startChCount = 0;
+  int        c       = 0;
+
+  newString();
+
+  while (((c = getNextYYChar(scanner)) != startCh || startChCount < 2) && c != 0) {
+    if (c == startCh) {
+      startChCount++;
+    } else {
+      startChCount = 0;
+    }
+
+    if (c == '\"') {
```

Comment:
I think 'addCharEscape' is unfortunately named. I'd prefer to rename it to 'addCharEscapeNonprint'. It's debateable whether or not much of the special handling below should go into a method named something like addCharEscape. But given that all of the strings represented inside the compiler are treated as C `"bla bla"` type strings, including appropriate backslashes, I think it'd be worthwhile to add a function for properly adding a character with appropriate escaping to such a string.

Reply:
But anyway if we keep the current structure for these cases, we know the character we're adding is printable, so shouldn't we just call addChar with it?